File: cloud/mdb/mdb-internal-api/internal/models/console/billing.go
package console

type BillingEstimate struct {
	// List of billing metrics of estimation
	Metrics []BillingMetric `json:"metrics" yaml:"metrics"`
}

type BillingMetric struct {
	// ID of the folder for metric
	FolderID string `json:"folder_id" yaml:"folder_id"`
	// Billing metric schema spec
	Schema string `json:"schema" yaml:"schema"`
	// Billing tags
	Tags BillingMetricTags `json:"tags" yaml:"tags"`
}

type BillingMetricTags struct {
	CloudProvider                   string   `json:"cloud_provider" yaml:"cloud_provider"`
	CloudRegion                     string   `json:"cloud_region" yaml:"cloud_region"`
	PublicIP                        int64    `json:"public_ip" yaml:"public_ip"`
	DiskTypeID                      string   `json:"disk_type_id" yaml:"disk_type_id"`
	ClusterType                     string   `json:"cluster_type" yaml:"cluster_type"`
	DiskSize                        int64    `json:"disk_size" yaml:"disk_size"`
	ResourcePresetID                string   `json:"resource_preset_id" yaml:"resource_preset_id"`
	ResourcePresetType              string   `json:"resource_preset_type" yaml:"resource_preset_type"`
	PlatformID                      string   `json:"platform_id" yaml:"platform_id"`
	Cores                           int64    `json:"cores" yaml:"cores"`
	CoreFraction                    int64    `json:"core_fraction" yaml:"core_fraction"`
	Memory                          int64    `json:"memory" yaml:"memory"`
	SoftwareAcceleratedNetworkCores int64    `json:"software_accelerated_network_cores" yaml:"software_accelerated_network_cores"`
	Roles                           []string `json:"roles" yaml:"roles"`
	Online                          int64    `json:"online" yaml:"online"`
	OnDedicatedHost                 int64    `json:"on_dedicated_host" yaml:"on_dedicated_host"`
	Edition                         string   `json:"edition" yaml:"edition"`
}

File: cloud/mdb/dbaas-internal-api-image/dbaas_internal_api/utils/billing.py
"""
Billing estimator helpers
"""

from typing import List, Union

from .types import ExistedHostResources, RequestedHostResources
from .validation import get_flavor_by_name


def generate_metric(
    host_spec: dict,
    resources: Union[RequestedHostResources, ExistedHostResources],
    cluster_type: str,
    roles: List[str],
    folder_id: str,
    schema: str = 'mdb.db.generic.v1',
    on_dedicated_host: bool = False,
) -> dict:
    """
    Generate billing metric (without usage part)
    """
    flavor = get_flavor_by_name(resources.resource_preset_id)

    return {
        'tags': {
            'public_ip': 1 if host_spec.get('assign_public_ip', False) else 0,
            'disk_type_id': resources.disk_type_id,
            'cluster_type': cluster_type,
            'disk_size': resources.disk_size,
            'resource_preset_id': resources.resource_preset_id,
            'platform_id': flavor.get('platform_id', ''),
            'cores': int(flavor.get('cpu_limit', 0)),
            'core_fraction': flavor.get('cpu_fraction', 100),
            'memory': flavor.get('memory_limit', 0),
            'software_accelerated_network_cores': flavor.get('io_cores_limit', 0),
            'roles': roles,
            'online': 1,
            'on_dedicated_host': 1 if on_dedicated_host else 0,
        },
        'folder_id': folder_id,
        'schema': schema,
    }
